### ISSUE TITLE
Current pingCommand is useless for macOS

### ISSUE THREAD

**Issue by beautyfree** [AlexGustafsson/homebridge-wol#108]
### Issue
Mac status is not working because in sleep state the ping command is always successful.

I could use a command like `ssh user@192.168.1.256 'if [[$ (pmset -g powerstate IODisplayWrangler | tail -1 | cut -c29) -lt 4]]; then; echo 0; else; exit 1; fi;'`
but the current execution of pingCommand doesn't make any sense to it

### Environment
* os & os version: macos catalina 10.15.5
* node version: v12.18.2
* homebridge version: 1.1.1
* homebridge-wol version: 4.2.1

### Configuration
My configuration looks like this:
```
    "accessories": [
        {
            "name": "Macbook Pro 2017",
            "ip": "192.168.1.256",
            "pingInterval": 45,
            "pingCommand": "ssh user@192.168.1.256 'if [[ $(pmset -g powerstate IODisplayWrangler | tail -1 | cut -c29) -lt 4 ]]; then; echo 0; else; exit 1; fi;'",
            "mac": "XX:XX:XX:XX:XX:XX",
            "broadcastAddress": "192.168.1.256",
            "wakeGraceTime": 10,
            "wakeCommand": "ssh user@192.168.1.256 caffeinate -u -t 300",
            "shutdownCommand": "ssh user@192.168.1.256 pmset sleepnow",
            "shutdownGraceTime": 15,
            "log": true,
            "logPinger": true,
            "debugLog": true,
            "returnEarly": false,
            "accessory": "NetworkDevice"
        }
    ],
```

**Comment by AlexGustafsson** [comment]
When you mean “current” do you mean the built-in ICMP ping? The plugin by default uses no ping command. The ping command you use is the recommended way of getting accurate pings off of macOS.

Also; your broadcast address looks off. In your configuration it looks like the default will do, so you can likely remove that config line.

**Comment by beautyfree** [comment]
@AlexGustafsson i mean that ping command everytime done successfully:
https://github.com/AlexGustafsson/homebridge-wol/blob/master/lib/pinger.js#L28

**Comment by AlexGustafsson** [comment]
I see. When a process spawned by node exits with a code different from 0, it will throw a runtime exception, which the try catch exception will catch and handle. Since the plugin doesn’t need to know exactly what code the process returned, or what its output was, there’s no need to refer to the process - which is why the line you’re referring to doesn’t seem to do much.  

If you’re experiencing that the plugin doesn’t care about the difference between the command `exit 0` and `exit 1`, then that’s a bug. If so, please try to create a minimum example of how to replicate it.

**Comment by beautyfree** [comment]
@AlexGustafsson oh ok, look likes it works. I just made mistake in command:
```
ssh user@192.168.1.256 'if [[ $(pmset -g powerstate IODisplayWrangler | tail -1 | cut -c29) -lt 4 ]]; then; exit 1; else; echo 1; fi;'
```

Works great for me

Maybe we need to add command like this to readme? I had to go around stackoverflow quite a lot to find an approach and apply it in this plugin :) I think it will help others to start using this plugin

**Comment by AlexGustafsson** [comment]
Great!

I’ve started collecting common configuration on the wiki, you’re very welcome to add it there. You’re also welcome to create a PR for the README and I’ll see to it that it’s merged. If you don’t have the time, I’ll add the comment the next time I take a look at the plugin.

Closing as the issue is solved.

**Comment by beautyfree** [comment]
@AlexGustafsson but i have interesting logs:
```
[7/20/2020, 11:10:17 PM] [Macbook Pro 2017] Got state request for "Macbook Pro 2017" (192.168.1.207): Offline
[7/20/2020, 11:10:20 PM] [Macbook Pro 2017] Attempting to poll state of "Macbook Pro 2017" (192.168.1.207) using "ssh"
[7/20/2020, 11:10:21 PM] [Macbook Pro 2017] Result of executing ping command for "Macbook Pro 2017" (192.168.1.207): online
```

i don't know why first line is offline

**Comment by AlexGustafsson** [comment]
The state request returns the state at that point in time. The last line is from the pinger, which will set the new state to online. Future state requests will return online. If you check further down in the log, you should see that the next request is in fact online.

**Comment by beautyfree** [comment]
@AlexGustafsson 
```
[7/20/2020, 11:26:23 PM] [Macbook Pro 2017] Got state request for "Macbook Pro 2017" (192.168.1.207): Offline
[7/20/2020, 11:26:26 PM] [Macbook Pro 2017] Got state request for "Macbook Pro 2017" (192.168.1.207): Offline
[7/20/2020, 11:26:47 PM] [Macbook Pro 2017] Attempting to poll state of "Macbook Pro 2017" (192.168.1.207) using "ssh"
[7/20/2020, 11:26:47 PM] [Macbook Pro 2017] Result of executing ping command for "Macbook Pro 2017" (192.168.1.207): online
[7/20/2020, 11:26:51 PM] [Macbook Pro 2017] Got state request for "Macbook Pro 2017" (192.168.1.207): Offline
[7/20/2020, 11:26:55 PM] [Macbook Pro 2017] Got state request for "Macbook Pro 2017" (192.168.1.207): Offline
```

**Comment by AlexGustafsson** [comment]
Yeah, that looks wrong. Reopening.

**Comment by AlexGustafsson** [comment]
Here's some more information regarding `pingCommand` for macOS: https://github.com/AlexGustafsson/homebridge-wol/issues/58

**Comment by AlexGustafsson** [comment]
I'm able to reproduce the issue. Investigating.

**Comment by AlexGustafsson** [comment]
I’ve found the bug. Hopefully I get the time to push a fix out today. I’ve also dockerized integration tests to simulate an actual setup with a MacBook etc. to make sure that issues like this are much easier to find in the future.

**Comment by AlexGustafsson** [comment]
Fixed in 4.2.2 which will be released shortly. If you still face the issue after upgrading, please comment here or reopen a new issue as soon as possible.
